ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -117,23 +116,13 @@ - (void)layoutSubviews
 
 - (void)setAnnotations:(NSMutableArray *)annotations
 {
-    _newAnnotations = annotations;
-    [self performSelector:@selector(updateAnnotations) withObject:nil afterDelay:0.5];
+    [self performSelector:@selector(updateAnnotations:) withObject:annotations afterDelay:0.5];
 }
 
-- (void)updateAnnotations {
-    if (_newAnnotations) {
-        if (_annotations.count) {
-            for (int i = 0; i < [_annotations count]; i++) {
-                [_map removeAnnotations: _annotations];
-            }
-            _annotations = nil;
-        }
-
-        _annotations = _newAnnotations;
-        for (int i = 0; i < [_newAnnotations count]; i++) {
-            [_map addAnnotation:_newAnnotations[i]];
-        }
+- (void)updateAnnotations:(NSMutableArray *) annotations {
+    for (int i = 0; i < [annotations count]; i++) {
+        [_annotations addObject:annotations[i]];
+        [_map addAnnotation:annotations[i]];
```

Comment:
`-[MGLMapView setAnnotations:]` handles both deleting existing annotations and adding all the new ones.